nltk/lm/api.py

```diff
@@ -16,15 +16,37 @@
 from nltk.lm.util import log_base2
 from nltk.lm.vocabulary import Vocabulary
 
-from itertools import accumulate
+try:
+    from itertools import accumulate, repeat
+except ImportError:
+    import operator
+
+    def accumulate(iterable, func=operator.add):
+        """Return running totals"""
+        # accumulate([1,2,3,4,5]) --> 1 3 6 10 15
+        # accumulate([1,2,3,4,5], operator.mul) --> 1 2 6 24 120
+        it = iter(iterable)
+        try:
+            total = next(it)
+        except StopIteration:
+            return
+        yield total
+        for element in it:
+            total = func(total, element)
+            yield total
+
+    def repeat(value, count):
+        """An iterator with the specified value repeated for count time"""
+        for _ in range(count):
+            yield (value)
 
 
 @add_metaclass(ABCMeta)
-class Smoothing:
+class Smoothing(object):
```

Comment:
Actually, here's something you can already do. Here and at several other points in the changeset it looks like you were basing your work on an older version of the `develop` branch.

Could you rebase your branch on the latest develop and re-upload that please?

Let me know if you need help rebasing.

Reply:
Actually, I patched this file on my local library first and just copied all of it into my fork. My branch is already based upon the latest version of `develop`.
If needed I can refractor the code a bit, but I don't think these differences are really breaking.

Reply:
Ah, I see.

It would be nice to remove all code that enables python 2 support, since NLTK is dropping support for that version and extra code is just that... extra.

I will comment on a few lines that I think are coming from an older version.